octobot/exchange_factory.py

```diff
@@ -58,6 +59,7 @@ def __init__(self, octobot, ignore_config=False):
                                                            backtesting_files=[os.getenv('BACKTESTING_FILE')])
                     await exchange_factory.create()
                     await init_exchange_chan_logger(exchange_factory.exchange_name)
+                    self.exchange_manager_list.append(exchange_factory.exchange_manager)
```

Comment:
Where do you use this attribute ?

Reply:
In interfaces, it's the entrypoint to retrieve trading data (using exchange_managers as keys in subsequent API calls)

Reply:
They doesn't use trading API ?

Reply:
We could but we need a way to differenciate exchange managers from a regular bot session from the ones of a backtesting bot session within a regular session => maybe we can add a method based on simulated exchanges and not simulated ones to get all instanciated exchange managers from the trading API. What do you think ?
To me it also makes sense that interfaces can access their octobot directly (required also for config and commands like stop/restart etc)

Reply:
Hum ok I see. I'm not really fond of a list of exchange manager outside of OctoBot-Trading and different list of exchange manager based on if they are simulated, in backtesting or something else.

I think your main problem here is that you can't request the trading API for multiple exchange_managers with the same exchange name ? Like if you have an exchange_manager running on binance and another that backtest on binance at the same time.

To this issue, we can imagine to keep using OctoBot-Trading API to request exchange_manager instances but with something like an ID. For example, when you start a backtesting you'll create a new instance of exchange_manager with a unique ID. This exchange_manager instance can be requested with the exchange name and the ID. 

What do you think about having a unique generated ID for each exchange manager that can be used to request the instance from OctoBot-Trading API ?

Reply:
Proposal in https://github.com/Drakkar-Software/OctoBot-Trading/pull/54

Reply:
That's a good idea and is solving almost all the problem. The only thing remaining is the way to identify which exchange managers are related to which "bot". I mean in backtesting form the web interface for example, how to get the exchange managers for the current backtesting session (without the ones from the normal "parent" session).

A way to keep track of these backtesting exchange managers would be to keep their ID somewhere, what do you think about storing them in the associated ```Backtesting``` instance ? Then they would be available through the OctoBot-Backtesting API

Reply:
Is it still required ? With the exchange ID implementation I don't think so.

Reply:
It's used to get the exchange managers from a given bot in interfaces, we can have a list of exchange ids instead but we need a way to identifie the exchange managers from the current bot (even in backtesting), exchange ids are better in my opinion.

Reply:
Ok be here you are making no difference between a backtesting bot and a classic one. And I don't think  we need a list of exchange_manager objects but only a list of exchange ids.